File: 3150-shortest-and-lexicographically-smallest-beautiful-string/3150-shortest-and-lexicographically-smallest-beautiful-string.py
class Solution:
    def shortestBeautifulSubstring(self, s: str, k: int) -> str:
        ans = ""
        max_len = 0

        for i in range(len(s)):
            cnt = 0
            for j in range(i, len(s)):
                if s[j] == '1':
                    cnt += 1
                    if cnt == k:
                        if max_len == 0:
                            max_len = j - i + 1
                        elif max_len > j - i + 1:
                            max_len = j - i + 1
                            break

        for i in range(len(s)):
            cnt = 0
            for j in range(i, len(s)):
                if s[j] == '1':
                    cnt += 1
                    if cnt == k:
                        if max_len == j - i + 1:
                            if not ans:
                                ans = s[i:i + max_len]
                            else:
                                ans = min(ans, s[i:i + max_len])

        return ans

File: 3150-shortest-and-lexicographically-smallest-beautiful-string/3150-shortest-and-lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string shortestBeautifulSubstring(string s, int k) {
        string ans = "";
        int len = 0;
        
        for (int i = 0; i < s.length(); i++) {
            int cnt = 0;
            for (int j = i; j < s.length(); j++) {
                if (s[j] == '1') {
                    cnt++;
                    if (cnt == k) {
                        if (len == 0) {
                            len = j - i + 1;
                        } else if (len > j - i + 1) {
                            len = j - i + 1;
                            break;
                        }
                    }
                }
            }
        }

        for (int i = 0; i < s.length(); i++) {
            int cnt = 0;
            for (int j = i; j < s.length(); j++) {
                if (s[j] == '1') {
                    cnt++;
                    if (cnt == k) {
                        if (len == j - i + 1) {
                            if (ans == "") {
                                ans = s.substr(i, len);
                            } else {
                                ans = min(ans, s.substr(i, len));
                            }
                        }
                    }
                }
            }
        }

        return ans;
    }
};